crates/rattler_index/src/main.rs

```diff
@@ -81,33 +84,17 @@ enum Commands {
         #[arg(value_parser = parse_s3_url)]
         channel: Url,
 
-        /// The endpoint URL of the S3 backend
-        #[arg(long, env = "S3_ENDPOINT_URL")]
-        endpoint_url: Option<Url>,
-
-        /// The region of the S3 backend
-        #[arg(long, env = "S3_REGION")]
-        region: Option<String>,
+        #[clap(flatten)]
+        credentials: rattler_s3::clap::S3CredentialsOpts,
```

Comment:
the CLI is still sort-of the same as before, right?

Reply:
Yes, except that when you specify any of these arguments, you must specify them all (except for the access key). E.g.

```
❯ rattler-index s3 --endpoint-url http://localhost s3://my-channel
error: the following required arguments were not provided:
  --region <REGION>
  
❯ rattler-index s3 --access-key-id http://localhost s3://my-channel
error: the following required arguments were not provided:
  --endpoint-url <ENDPOINT_URL>
  --region <REGION>
  --secret-access-key <SECRET_ACCESS_KEY>
```